src/Functions/intExp2.cpp

```diff
@@ -23,9 +24,22 @@ struct IntExp2Impl
     static ResultType apply([[maybe_unused]] A a)
     {
         if constexpr (is_big_int_v<A>)
+        {
             throw DB::Exception(ErrorCodes::NOT_IMPLEMENTED, "intExp2 not implemented for big integers");
+        }
         else
+        {
+            if constexpr (std::is_floating_point_v<A>)
+            {
+                if (std::isnan(a))
+                    throw Exception(ErrorCodes::BAD_ARGUMENTS, "intExp2 must not be called with nan");
+                if (a < 0)
+                    return 0;
+                if (a >= 64)
+                    return std::numeric_limits<UInt64>::max();
```

Comment:
We should make those comparisons here because otherwise `static_cast<int>()` can stop with for example
```
/Functions/intExp2.cpp: runtime error: 1e+20 is outside the range of representable values of type 'int'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /Functions/intExp2.cpp
```